### PR TITLE
[Documentation] Available queryParam types

### PR DESCRIPTION
**Pull request by lifeart**
For now, only 'boolean' query params type present in documentation.

This pr brings all available types for queryParam in documentation.

https://github.com/emberjs/ember.js/blob/6fc89cdf13124d88b8ae6adf99bb02a8c0cdf508/packages/ember-routing/lib/system/router.js#L732

### CODE REVIEW

File: packages/ember-routing/lib/ext/controller.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ ControllerMixin.reopen({
     `this.category` and `this.page`.
     By default, Ember coerces query parameter values using `toggleProperty`.
     This behavior may lead to unexpected results.
+    Available queryParam types: `boolean`, `number`, `array`
```
Comment (btecu) [inline packages/ember-routing/lib/ext/controller.js]: Also `string`?

Comment (lifeart) [reply packages/ember-routing/lib/ext/controller.js]: @btecu nope, https://github.com/emberjs/ember.js/blob/6fc89cdf13124d88b8ae6adf99bb02a8c0cdf508/packages/ember-routing/lib/system/router.js#L732

Comment (btecu) [reply packages/ember-routing/lib/ext/controller.js]: Isn't `string` the default?

Comment (lifeart) [reply packages/ember-routing/lib/ext/controller.js]: it can be `null`, `undefined`, or `string`

Comment (lifeart) [reply packages/ember-routing/lib/ext/controller.js]: in happy path - `string`

**Comment by locks** [comment]
Can you update this PR? 🙏

**Comment by locks** [comment]
@lifeart hi! do you have some time to handle the conflicts?

**Comment by lifeart** [comment]
@locks yeah, I'm in

**Comment by lifeart** [comment]
I will open another PR )

**Pull request closed without merging**